org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/parser/CompUtil.java

```diff
@@ -262,42 +263,92 @@ private static CompModule parseRecursively(List<Object> seenDollar, Map<String,S
 
         // Here, we recursively open the included files
         for (Open x : u.getOpens()) {
-            String cp = Util.canon(computeModulePath(u.getModelName(), filename, x.filename)), content = fc.get(cp);
-            try {
-                if (content == null) {
-                    content = loaded.get(cp);
-                }
-                if (content == null) {
-                    content = fc.get(x.filename);
-                    if (content != null)
-                        cp = x.filename;
+            String cp_als = Util.canon(computeModulePath(u.getModelName(), filename, x.filename));
```

Comment:
This would also be a lot easier to read if it was delegated to a separate object. Please do not use statics in that solution.